com.unity.shadergraph/Editor/Data/Graphs/Vector2MaterialSlot.cs

```diff
@@ -20,6 +20,9 @@ class Vector2MaterialSlot : MaterialSlot, IMaterialSlotHasValue<Vector2>
         [SerializeField]
         string[] m_Labels; // this can be null, which means fallback to k_LabelDefaults
 
+        [SerializeField]
+        bool m_Integer = false;
```

Comment:
Yep, this serialize field can be removed. 

I only added it to Vector2 because that's the only use case I have for the Gather node. I figured it's trivial enough to extend that if another node calls for it. 

Same for Gather and CalculateLevelOfDetail - those could be extended to cubemaps/arrays too, but I had no requirement for it in the current scope of work.

Reply:
Removed.